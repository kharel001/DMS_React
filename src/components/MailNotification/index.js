import React from 'react';

import NotificationItem from './NotificationItem';
import {notifications} from './data';
import CustomScrollbars from '../../util/CustomScrollbars';

const MailNotification = () => {

  return (
    <CustomScrollbars className="messages-list scrollbar" style={{height: 280}} 
    data-test="customScrollComp">
      <ul className="list-unstyled">
        {notifications.map((notification, index) => <NotificationItem key={index}
                                                                      notification={notification}
                                                                      data-test="notificationItemComp"/>)}
      </ul>
    </CustomScrollbars>
  )
};

export default MailNotification;

