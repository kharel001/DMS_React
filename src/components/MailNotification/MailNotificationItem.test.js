import React from 'react';
import MailNotification from './index';
import {
    configure,
    shallow,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})

const setup = (props = {}) => {
    return shallow( < MailNotification
     {...props}  />
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}

{/* it("Custom Scrollable render", () => {
        let wrapper = setup(dummy_data);
        // expect(wrapper).toMatchSnapshot();
}) */}
it("Snapshot MailNotification", () => {
    let wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    // const span2= wrapper2.find('div')
    // expect(span2).toHaveLength(5); //doesn't work as expected
    // expect(img1.props("src")).toEqual("https://via.placeholder.com/150x150")
    
})



it("Components MailNotification", () => {
    let wrapper = setup();
    let customScrollComp = findByTestAttr(wrapper, "customScrollComp");
    expect(customScrollComp).toHaveLength(1);
    let notificationItemComp = findByTestAttr(wrapper, "notificationItemComp");
    expect(notificationItemComp).toHaveLength(5);
})