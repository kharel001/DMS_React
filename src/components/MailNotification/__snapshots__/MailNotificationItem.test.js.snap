// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot MailNotification 1`] = `ShallowWrapper {}`;
